File: Lab1/golang/main.go
package main

import (
	"fmt"
	"math"
)

func main() {
	var a, b, c, r float64 //a, b, c -- кути трикутника, r -- радіус описаного кола
	fmt.Print(">>> Вкажіть градусну міру кута a: ")
	_, _ = fmt.Scan(&a) //Зчитування
	fmt.Print(">>> Вкажіть градусну міру кута b: ")
	_, _ = fmt.Scan(&b)
	fmt.Print(">>> Вкажіть градусну міру кута c: ")
	_, _ = fmt.Scan(&c)
	fmt.Print(">>> Вкажіть радіус описаного кола: ")
	_, _ = fmt.Scan(&r)
	if a+b+c == 180 && a > 0 && b > 0 && c > 0 && r > 0 { //перевірка існування трикутника
		side1 := sideCalculate(r, a)
		side2 := sideCalculate(r, b)
		side3 := sideCalculate(r, c)
		fmt.Println(">>> Сторони трикутника рівні ", side1, ", ", side2, " і ", side3)
	} else {
		fmt.Println(">>> Сума кутів не рівна 180 :(")
	}
}
func toRadian(angle float64) float64 {
	return (angle * math.Pi) / 180 // Функція переведення в радіани
}
func sideCalculate(r, a float64) float64 {
	return 2 * r * math.Sin(toRadian(a)) // Функція розрахунку сторони за радіусом описаного кола та кутом (т. синусів)
}

File: Lab1/javaScript/main.js
function calculate() {
    let a = parseFloat(document.getElementById("inp1").value); //Зчитування міри кута а
    let b = parseFloat(document.getElementById("inp2").value); //Зчитування міри кута b
    let c = parseFloat(document.getElementById("inp3").value); //Зчитування міри кута c
    let r = parseFloat(document.getElementById("inp4").value); //Зчитування міри радіуса вписаного кола
    if (a + b + c === 180) { //перевірка існування трикутника
        let side1 = sideCalculate(a, r);
        let side2 = sideCalculate(b, r);
        let side3 = sideCalculate(c, r);
        document.getElementById("result").innerText = "Сторони трикутника рівні " + side1 + ", " + side2 + " і " + side3;
    } else {
        document.getElementById("result").innerText = "Сума кутів не рівна 180 :(";
    }
}

function sideCalculate(a, r) { // Функція розрахунку сторони за радіусом вписаного кола та кутом (т. синусів)
    return 2 * r * Math.sin((toRadian(a)));
}

function toRadian(angle) { // Функція переведення в радіани
    return (angle * Math.PI) / 180;
}